tactics/equality.ml

```diff
@@ -988,7 +984,7 @@ let discrimination_pf env sigma e (t,t1,t2) discriminator lbeq =
   let sigma, i           = build_coq_I sigma in
   let sigma, absurd_term = build_coq_False sigma in
   let eq_elim, eff       = ind_scheme_of_eq lbeq in
-  let sigma, eq_elim = Evd.fresh_global (Global.env ()) sigma eq_elim in
+  let sigma, eq_elim = Evd.fresh_global env sigma eq_elim in
```

Comment:
Nope, I think you should rather change build_coq_I, build_coq_False etc to use pf_constr_of_global instead, it's currently using the lower level Evarutil.new_global.